DemoApp/CreateChatViewController.swift

```diff
@@ -0,0 +1,251 @@
+        cell.nameLabel.text = user.name
+        
+        if let lastActive = user.lastActiveAt {
+            cell.descriptionLabel.text = "Last seen: " + formatter.string(from: lastActive)
+        } else {
+            cell.descriptionLabel.text = "Never seen"
+        }
+        
+        if selectedUserIds.contains(user.id) {
+            cell.accessoryImageView.image = UIImage(systemName: "checkmark.circle.fill")
+        } else {
+            cell.accessoryImageView.image = nil
+        }
+        
+        cell.user = user
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath) as! UserCredentialsCell
+        
+        if cell.accessoryImageView.image == nil {
+            // Select user
+            cell.accessoryImageView.image = UIImage(systemName: "checkmark.circle.fill")
+            let token = UISearchToken(
+                icon: cell.avatarView.image?.resized(to: .init(width: 20, height: 20)),
+                text: cell.user?.name ?? cell.user?.id ?? "NoName"
+            )
+            token.representedObject = cell.user
+            searchField.replaceTextualPortion(of: searchField.textualRange, with: token, at: searchField.tokens.count)
+        } else {
+            // Deselect user
+            cell.accessoryImageView.image = nil
+            if let tokenIndex = searchField.tokens.firstIndex(where: { ($0.representedObject as? ChatUser)?.id == cell.user?.id }) {
+                searchField.removeToken(at: tokenIndex)
+            }
+        }
+        
+        tableView.deselectRow(at: indexPath, animated: true)
+        createGroupStack.isHidden = searchField.hasText
+    }
+}
+
+extension CreateChatViewController: UserSearchControllerDelegate {
+    func userSearchController(_ controller: UserSearchController, didFetchUsers users: [ChatUser]) {
+        noMatchView.isHidden = !users.isEmpty
+        activityIndicator.stopAnimating()
+        tableView.reloadData()
+    }
+    
+    func userSearchController(_ controller: UserSearchController, didUpdateUsers changes: [ListChange<ChatUser>]) {
+        tableView.beginUpdates()
+        
+        for change in changes {
+            switch change {
+            case let .insert(_, index: index):
+                tableView.insertRows(at: [index], with: .automatic)
+            case let .move(_, fromIndex: fromIndex, toIndex: toIndex):
+                tableView.moveRow(at: fromIndex, to: toIndex)
+            case let .update(_, index: index):
+                tableView.reloadRows(at: [index], with: .automatic)
+            case let .remove(_, index: index):
+                tableView.deleteRows(at: [index], with: .automatic)
+            }
+        }
+        
+        tableView.endUpdates()
+        
+        noMatchView.isHidden = !users.isEmpty
+    }
+}
+
+extension UIImage {
+    func resized(to targetSize: CGSize) -> UIImage {
```

Comment:
Since we call this for every cell without cashing it, my guess would be that this is actually worse than just putting full images into the avatars :))

Reply:
We call this for search tokens (so only selected cells)